src/containers/cooperation-details/add-course-modal-modal/AddCourseTemplateModal.tsx

```diff
@@ -64,21 +60,24 @@ const AddCourseTemplateModal: FC<AddCourseTemplateModalProps> = ({
   }
 
   const getUserData = useCallback(
-    () => userService.getUserById(userId, userRole as UserRole),
+    () => userService.getUserByIdWithBaseService(userId, userRole as UserRole),
     [userId, userRole]
   )
 
-  const { loading: userLoading, response: user } =
-    useAxios<UserResponse | null>({
-      service: getUserData,
-      defaultResponse: null
-    })
-
-  const getCourses = useCallback(() => CourseService.getCourses(), [])
+  const { isLoading: userLoading, data: userData = null } = useQuery({
```

Comment:
`userData` will be undefined while fetching, so you don't need to add `null` as default value
```suggestion
  const { isLoading: userLoading, data: userData } = useQuery({
```

Reply:
In the CoursesFilters component, UserResponse | null is expected, not UserResponse | undefined
![image](https://github.com/user-attachments/assets/f7317b02-acb7-4fc1-990c-10ed4f0643ec)

Reply:
I got it, you can leave it as it is